controllers/chat/sendSignedMessage.js

```diff
@@ -131,8 +132,44 @@ const sendSignedMesage = async (req, res) => {
       });
     }
 
-    const chat = await channel.sendMessage(baseMessage);
+    const chat = await channel.sendMessage(baseMessage, {skip_push: true});
+    let notificationPayload = {
+      title: `${senderInfo?.user?.username}`,
+      body: `${message.substring(0, 100)}`
+    };
+    let dataPayload = {
+      channel_id: channelId,
+      user_id: req.userId,
+      messages_id: chat.message.id,
+      message,
+      message_schema: currentMessageType,
+      type: 'message.new',
+      status: 'sent',
+      is_big_message: 'false',
+      attachment: JSON.stringify(attachments) ?? '',
+      created_at: chat.message.created_at,
+      is_annoymous: 'false'
+    };
+    const dataSizeInBytes = Buffer.byteLength(JSON.stringify(dataPayload), 'utf8');
+    const dataSizeInKilobytes = dataSizeInBytes / 1024;
+    if (dataSizeInKilobytes > 4) {
+      dataPayload.is_big_message = 'true';
+      dataPayload.attachment = '';
+    }
 
+    channelMember.forEach((member) => {
+      let payload = {
+        notification: notificationPayload,
+        data: {
+          ...dataPayload,
+          is_annoymous: member.is_anonymous.toString()
+        }
+      };
+      BetterSocialCore.fcmToken.sendChatNotification(member.user_id, payload, {
+        priority: 'high',
+        content_available: true
+      });
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code sends a notification to each member of the channel. However, there's no error handling for the `sendChatNotification` function. If sending a notification fails for one member, it could stop the notifications for the remaining members. Consider adding error handling or using Promise.allSettled to ensure all notifications are sent even if some fail.
```javascript
const notificationPromises = channelMember.map((member) => {
  let payload = {
    notification: notificationPayload,
    data: {
      ...dataPayload,
      is_annoymous: member.is_anonymous.toString()
    }
  };
  return BetterSocialCore.fcmToken.sendChatNotification(member.user_id, payload, {
    priority: 'high',
    content_available: true
  }).catch((error) => {
    console.error(`Failed to send notification to user ${member.user_id}:`, error);
  });
});

await Promise.allSettled(notificationPromises);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->